ground/src/main/java/com/google/android/ground/persistence/remote/firebase/protobuf/MessageLiteReflectionExt.kt

```diff
@@ -148,3 +149,12 @@ private fun KProperty<*>.isMessageFieldProperty(): Boolean =
   name.endsWith("_") && !name.startsWith(BIT_FIELD_PROPERTY_PREFIX)
 
 fun String.toSnakeCase() = replace(Regex("[A-Z]")) { "_${it.value}" }.lowercase()
+
+private fun getEnumValues(enumClass: KClass<out Enum<*>>): List<Enum<*>> =
+  enumClass.java.enumConstants.toList()
+
+@Suppress("UNCHECKED_CAST")
+fun <T : EnumLite> KClass<T>.findByNumber(number: Int): T? {
+  require(isSubclassOf(Enum::class))
+  return getEnumValues(this as KClass<Enum<*>>).find { (it as EnumLite).number == number } as? T
```

Comment:
Not sure if this is the same number, but `getNumber()` seems to be implemented on the `EnumLite` interface, which should be more persistent between library upgrades.

https://protobuf.dev/reference/java/api-docs/com/google/protobuf/Internal.EnumLite.html#getNumber()

Reply:
Yes, `.number` is the Kotlinic way of calling `.getNumber()`.